### PR TITLE
fix: Use bcrypt for argo password, don't create cw log group

### PR DESCRIPTION
**Pull request by Zvikan**
### What does this PR do?
- adds custom bcrypt provider for argo example
Using custom bcrypt provider instead of the Hashicorp default one because the default generates a diff for every TF plan, leading to issues like https://github.com/aws-ia/terraform-aws-eks-blueprints/issues/1113 where the resources appears as changed every time. as mentioned in the [docs](https://registry.terraform.io/providers/viktorradnai/bcrypt/latest/docs/resources/hash): 
```
Unlike the built-in bcrypt function, this resource will not generate a diff unless the cleartext value changes.
```
- disables fluentbit CW log group creation by Terraform for Argo example
🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

### Motivation

- Related to https://github.com/aws-ia/terraform-aws-eks-blueprints/issues/1113
- bcrypt required by Argo as the password has to bcrypt
- fluentbit currently tries to create CW log group like Terraform, and this causes terraform to fail sometimes with "log group with the same name already exists", so for now, disabling TF from creating the log group and letting the addon itself to do it (confirmed IRSA permissions created log group in cloud trail).

### More

- [ ] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [ ] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [ ] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs require a new example and/or doc page. In general:
- Use an existing example when possible to demonstrate a new addons usage
- A new docs page under `docs/add-ons/*` is required for new a new addon

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

**Review by askulkarni2 — COMMENTED** [review]
Why not use the built-in [bcrypt()](https://developer.hashicorp.com/terraform/language/functions/bcrypt)?

File: examples/gitops/argocd/main.tf [hunk 1]
```diff
@@ -132,6 +137,12 @@ resource "random_password" "argocd" {
   override_special = "!#$%&*()-_=+[]{}<>:?"
 }
 
+# Argo requires the password to be bcrypt, we use custom provider of bcrypt,
+# as the default bcrypt function generates diff for each terraform plan
```
Comment (Zvikan) [inline examples/gitops/argocd/main.tf:141]: @askulkarni2 reason why custom bcrypt mentioned here

Comment (askulkarni2) [reply examples/gitops/argocd/main.tf:141]: D'oh apologies!

**Review by askulkarni2 — APPROVED** [review]
LGTM!

**Pull request merged**